bigbite/image-comparison
# Replace image URI attribute with state

## Description
This PR removes the `url` attribute, which is only used in the editor.  
It then adds a stateful approach to loading the correct image URI in the editor.  

This will allow for easier production of sample content for usage via such things as the WP Playground.

## Change Log
- Replace image URI attribute with state

## Steps to test
1. before checking out this PR, insert an Image Comparison block, and add two images. Save
2. checkout this PR
3. run `npm i`
4. run `npm run build:prod`
5. reload the post containing the Image Comparison block you added in step 1
6. the block should load correctly, and should still display the two images you inserted
7. Save and reload the post again
8. the block should still load correctly, and should still display the two images you inserted
9. select an image within the block, and change the media size
10. the selected size should load in correctly
11. Save and reload the post again
12. the block should still load correctly, and should still display the two images you inserted, at the sizes you chose
13. replace one of the images
14. the replacement image should load in correctly
15. Save and reload the post again
16. the block should still load with the correct media

## Screenshots/Videos
~

## Checklist:
